components/TextFields/src/MDCTextInputControllerBase.m

```diff
@@ -211,34 +211,50 @@ - (instancetype)init {
 - (instancetype)initWithCoder:(NSCoder *)aDecoder {
   self = [super init];
   if (self) {
-    _activeColor = [aDecoder decodeObjectForKey:MDCTextInputControllerBaseActiveColorKey];
-    _borderFillColor = [aDecoder decodeObjectForKey:MDCTextInputControllerBaseBorderFillColorKey];
-    _characterCounter = [aDecoder decodeObjectForKey:MDCTextInputControllerBaseCharacterCounterKey];
-    _disabledColor = [aDecoder decodeObjectForKey:MDCTextInputControllerBaseDisabledColorKey];
-    _errorColor = [aDecoder decodeObjectForKey:MDCTextInputControllerBaseErrorColorKey];
+    _activeColor = [aDecoder decodeObjectOfClass:[UIColor class]
+                                          forKey:MDCTextInputControllerBaseActiveColorKey];
+    _borderFillColor = [aDecoder decodeObjectOfClass:[UIColor class]
+                                              forKey:MDCTextInputControllerBaseBorderFillColorKey];
+    _characterCounter =
+        [aDecoder decodeObjectOfClass:[NSObject<MDCTextInputCharacterCounter> class]
+                               forKey:MDCTextInputControllerBaseCharacterCounterKey];
+    _disabledColor = [aDecoder decodeObjectOfClass:[UIColor class]
+                                            forKey:MDCTextInputControllerBaseDisabledColorKey];
+    _errorColor = [aDecoder decodeObjectOfClass:[UIColor class]
+                                         forKey:MDCTextInputControllerBaseErrorColorKey];
     _expandsOnOverflow = [aDecoder decodeBoolForKey:MDCTextInputControllerBaseExpandsOnOverflowKey];
     _floatingEnabled = [aDecoder decodeBoolForKey:MDCTextInputControllerBaseFloatingEnabledKey];
     _floatingPlaceholderNormalColor =
-        [aDecoder decodeObjectForKey:MDCTextInputControllerBaseFloatingPlaceholderNormalColorKey];
+        [aDecoder decodeObjectOfClass:[UIColor class]
+                               forKey:MDCTextInputControllerBaseNormalColorKey];
```

Comment:
needs to be `MDCTextInputControllerBaseFloatingPlaceholderNormalColorKey`

Reply:
Thanks for catching these! It's not easy.